crates/viewer/re_space_view_dataframe/src/space_view_class.rs

```diff
@@ -167,23 +124,25 @@ mode sets the default time range to _everything_. You can override this in the s
     ) -> Result<(), SpaceViewSystemExecutionError> {
         re_tracing::profile_function!();
 
-        let settings = ViewProperty::from_archetype::<archetypes::DataframeViewMode>(
-            ctx.blueprint_db(),
-            ctx.blueprint_query,
-            query.space_view_id,
-        );
-
-        let mode =
-            settings.component_or_fallback::<components::DataframeViewMode>(ctx, self, state)?;
-
-        // update state
-        let state = state.downcast_mut::<DataframeViewState>()?;
-        state.mode = mode;
-
-        match mode {
-            components::DataframeViewMode::LatestAt => latest_at_table_ui(ctx, ui, query),
+        let view_query = super::view_query::Query::try_from_blueprint(ctx, query.space_view_id)?;
+        let timeline_name = view_query.timeline_name(ctx);
+        let query_mode = view_query.mode(ctx);
+
+        let Some(timeline) = ctx
+            .recording()
+            .timelines()
+            .find(|t| t.name() == &timeline_name)
+        else {
+            re_log::warn_once!("Could not find timeline {:?}.", timeline_name.as_str());
+            //TODO(ab): we should have an error for that
+            return Ok(());
+        };
```

Comment:
The `SpaceViewSystemExecutionError` return type is a bit rigid for the various error situation one might encounter.

Reply:
We can add a `SpaceViewSystemExecutionError::Custom` maybe?